src/execution/riscv64/simulator-riscv64.cc

```diff
@@ -3171,29 +3167,17 @@ void Simulator::Execute() {
   // Get the PC to simulate. Cannot use the accessor here as we need the
   // raw PC value and not the one used as input to arithmetic instructions.
   int64_t program_counter = get_pc();
-  if (::v8::internal::FLAG_stop_sim_at == 0) {
-    // Fast version of the dispatch loop without checking whether the
-    // simulator should be stopping at a particular executed instruction.
-    while (program_counter != end_sim_pc) {
-      Instruction* instr = reinterpret_cast<Instruction*>(program_counter);
-      icount_++;
```

Comment:
The original implementation tests `FLAG_stop_sim_at == 0` at the top level to speed-up the common path. I think it is better to preserve this structure (although we have to replicate the inner while loop twice.